### ISSUE TITLE
Adjust user model to include properties for Property Managers

### ISSUE THREAD

**Issue by DaveTrost** [codeforpdx/dwellingly-app#195]
**Expected Behavior**
When GET /user/:id is called for a Property Manager, the properties they manage and the tenants that belong to those properties should be returned

pseudo-code from Hugh on how to accomplish this:
```
if role == "PROPERTY MANAGER":
 return user.json w/ properties
```

**Comment by dbridenbeck** [comment]
For what its worth, below is the dummy data I'm currently using. If the returned data resembled this, it would be awesome. If it can't, I'm sure we can find a way on the front end to still make it work.

```
export const PROPERTY_MANAGER_DATA = [
	{
		id: "1",
		firstName: "Jim",
		lastName: "Oliver",
		fullName: "Jim Oliver",
		phone: "(503) 123-1234",
		lastUsage: "9/12/2019",
		status: "Active",
		email: "john.oliver@propertiespdx.org",
		properties: [
			{
				name: "Jim's Mansion",
				streetAddress: "1234 Burnside Street",
				city: "Portland",
				state: "OR",
				zip: "97211",
			},
			{
				name: "Meerkat Mansion",
				streetAddress: "4552 Burnside Street",
				city: "Portland",
				state: "OR",
				zip: "97231",
			},
		],
		tenants: [
			{
				name: "Brendan Smith",
				property: "The Magnolia",
				unit: "Unit #10",
				phone: "(503) 123-1234",
			},
			{
				name: "Maggie Taylor",
				property: "The Magnolia",
				unit: "Unit #2",
				phone: "(503) 123-1234",
			},
			{
				name: "Brock Hampton",
				property: "Gibon Gardens",
				unit: "Unit #8",
				phone: "(503) 123-1234",
			},
			{
				name: "Jason Nyguen",
				property: "Gibon Gardens",
				unit: "Unit #12",
				phone: "(503) 123-1234",
			},
		],
	},
	{
		id: "2",
		firstName: "Bob",
		lastName: "Smith",
		fullName: "Bob Smith",
		status: "Pending",
		phone: "(503) 123-1434",
		email: "bob.smith@propertiespdx.org",
		lastUsage: "9/12/2019",
		properties: [
			{
				name: "Bob's Mansion",
				streetAddress: "1224 Hill Street",
				city: "Portland",
				state: "OR",
				zip: "97777",
			},
			{
				name: "Orange Courts",
				streetAddress: "2222 Dover Street",
				city: "Portland",
				state: "OR",
				zip: "97341",
			},
		],
		tenants: [
			{
				name: "Dougie Houser",
				property: "The Bluffs",
				unit: "Unit #1000",
				phone: "(503) 543-2100",
			},
			{
				name: "Reginald Stevens",
				property: "The Bluffs",
				unit: "Unit #22",
				phone: "(503) 555-5555",
			},
			{
				name: "Stacy Douglas",
				property: "The West Point",
				unit: "Unit #50",
				phone: "(503) 555-4444",
			},
		],
	},
];
```

**Comment by sokampdx** [comment]
@dbridenbeck Do you need the id for properties? I remember they mentioned something about that in the meeting.

Also, do you need the tenants list under the property that they lived in? 

Finally, after this [issue](https://github.com/codeforpdx/dwellingly-app/issues/236), we would allow multiples tenants under one property.

**Comment by dbridenbeck** [comment]
@sokampdx I think you're spot on in your assumptions. Let's add the ID's for properties **and** tenants. Also having tenants under properties makes logical sense and we can refactor the frontend to match that data structure. And yes, having more than one tenant in a property is . . . important :)

**Comment by sokampdx** [comment]
@dbridenbeck
Additional questions base on your dummy data.
I see the following that are not in the models currently.	Would this be additional ticket later?	
```
		lastUsage: "9/12/2019",
		status: "Active",
```

Do you need other information from the models? e.g. archived, dateAdded, role... etc.

**Comment by dbridenbeck** [comment]
@sokampdx Ah interesting. If lastUsage and status don't exist on the model, we should create a ticket for those. Not sure based on the scope if it should be one ticket or two, I'll leave that up to you and the other backend experts :)

I don't need any further information from the models. I scanned the Figma file and can't find instances where we would, but I could be wrong. If I am, perhaps we just open a new ticket?

**Comment by sokampdx** [comment]
@dbridenbeck when the front end request info for an user by user id and the backend return a json of either a vanilla user or a property manager, using what we have right now, fe would need to peel out the json to find if there is a property key to be able distinguish the different user roles. Would giving fe the role eliminate that uncertainty?

**Comment by dbridenbeck** [comment]
@sokampdx great questions and good thinking, but I'm pretty sure that GET /api/user/:id already returns the "role" property in the response. Below is my current thinking of the fe implementation, let me know if you see any issues with it or if there's something I'm missing that would need to change on the backend. Thanks for being so thorough with this ticket! :)

1. the view /manage/managers (page 16 on the figma docs) would populate the table with data by hitting POST /api/users/role with the following body:
```
{
	"userrole": "PROPERTY MANAGER"
}
```

In that response, we would need each property manager's basic information, their properties & property id's. Sample response below (I now realize that the current response from the backend does not match the example below. To resolve this ticket, I think we need to solve this challenge as well):
```
{
  "firstName": "Dave",
  "lastName": "Smith",
  "properties": [
    {
      "propertyName": "Meercat Manor",
      "propertyId": 4
    },
    {
      "propertyName": "Doogan's Courts",
      "propertyId": 5
    },
  ],
  "email": "dave.smith@gmail.com",
  "status": "active",
  "lastUsage": "06/02/2019"
}
```

2. then, when a property manager's name is clicked in the table (see figma file pg. 16), they would be routed to the view /manage/manager/:id (page 17 of figma file), and the id from the URL would be used to GET /api/user/:id, which, once this ticket is done, would return the property manager's basic details (first name, last name, phone, email), properties, and tenants under each property. Currently when I hit that route, it provides this response, which already provides the user role:
```
{
    "id": "7",
    "firstName": "frank",
    "lastName": "georgely",
    "email": "george.bigman@gmail.com",
    "role": "PROPERTY MANAGER",
    "archived": false
}
```

I hope this context is helpful, and please let me know if I'm missing something here :)

**Comment by sokampdx** [comment]
Thanks for the detailed explanation in your reply. The figma page number really helps! 
For figma page 16, I will add the associated properties for the response on POST /users/role. 
For figma page 17, the layout of the UI encouraged separating the list for properties and tenants, which is what you originally proposed in your dummy data. I will do that as well.

**Comment by sokampdx** [comment]
Pull request: https://github.com/codeforpdx/dwellinglybackend/pull/55/files

Testing description:

setup:

- Using postman, post at /login with the credential and setup the bearer token under authorization
- ran the following test:

Test case against backend api:

1. test non property manager user id
GET 127.0.0.1:5000/api/user/1
RESPONSE
`
{
    "id": 1,
    "firstName": "user1",
    "lastName": "tester",
    "email": "user1@dwellingly.org",
    "role": "admin",
    "archived": false
}
`
2. test property manager id with no tenants
GET 127.0.0.1:5000/api/user/4
RESPONSE
`
{
    "id": 4,
    "firstName": "Mr.",
    "lastName": "Sir",
    "email": "MisterSir@dwellingly.org",
    "role": "property-manager",
    "archived": false,
    "properties": [
        {
            "id": 2,
            "name": "Meerkat Manor",
            "address": "Privet Drive",
            "city": "Portland",
            "state": "OR",
            "zipcode": "97207",
            "propertyManager": 4,
            "tenants": 6,
            "dateAdded": "2020-04-12",
            "archived": false
        }
    ],
    "tenants": []
}
`
3. test property manger with tenants
GET 127.0.0.1:5000/api/user/5
RESPONSE
`
{
    "id": 5,
    "firstName": "Gray",
    "lastName": "Pouponn",
    "email": "user3@dwellingly.org",
    "role": "property-manager",
    "archived": false,
    "properties": [
        {
            "id": 1,
            "name": "test1",
            "address": "123 NE FLanders St",
            "city": "Portland",
            "state": "OR",
            "zipcode": "97207",
            "propertyManager": 5,
            "tenants": 3,
            "dateAdded": "2020-04-12",
            "archived": false
        },
        {
            "id": 3,
            "name": "The Reginald",
            "address": "Aristocrat Avenue",
            "city": "Portland",
            "state": "OR",
            "zipcode": "97207",
            "propertyManager": 5,
            "tenants": 4,
            "dateAdded": "2020-04-12",
            "archived": false
        }
    ],
    "tenants": [
        {
            "id": 3,
            "firstName": "Starvin",
            "lastName": "Artist",
            "phone": "123-123-1111",
            "propertyID": 2,
            "propertyName": "Meerkat Manor",
            "propertyAddress": "Privet Drive",
            "propertyTenants": 6,
            "staff": null
        }
    ]
}
`
4. test userrole property manager return a list of property with user info
POST 127.0.0.1:5000/api/users/role
BODY
`
{
    "userrole": "property-manager" 
}
`
RESPONSE
`
{
    "users": [
        {
            "id": 4,
            "firstName": "Mr.",
            "lastName": "Sir",
            "email": "MisterSir@dwellingly.org",
            "role": "property-manager",
            "archived": false,
            "properties": [
                {
                    "id": 2,
                    "name": "Meerkat Manor",
                    "address": "Privet Drive",
                    "city": "Portland",
                    "state": "OR",
                    "zipcode": "97207",
                    "propertyManager": 4,
                    "tenants": 6,
                    "dateAdded": "2020-04-12",
                    "archived": false
                }
            ]
        },
        {
            "id": 5,
            "firstName": "Gray",
            "lastName": "Pouponn",
            "email": "user3@dwellingly.org",
            "role": "property-manager",
            "archived": false,
            "properties": [
                {
                    "id": 1,
                    "name": "test1",
                    "address": "123 NE FLanders St",
                    "city": "Portland",
                    "state": "OR",
                    "zipcode": "97207",
                    "propertyManager": 5,
                    "tenants": 3,
                    "dateAdded": "2020-04-12",
                    "archived": false
                },
                {
                    "id": 3,
                    "name": "The Reginald",
                    "address": "Aristocrat Avenue",
                    "city": "Portland",
                    "state": "OR",
                    "zipcode": "97207",
                    "propertyManager": 5,
                    "tenants": 4,
                    "dateAdded": "2020-04-12",
                    "archived": false
                }
            ]
        }
    ]
}
`

5. test userrole admin
POST 127.0.0.1:5000/api/users/role
BODY
`
{
    "userrole": "admin" 
}
`
RESPONSE
`
{
    "users": [
        {
            "id": 1,
            "firstName": "user1",
            "lastName": "tester",
            "email": "user1@dwellingly.org",
            "role": "admin",
            "archived": false,
            "properties": []
        },
        {
            "id": 2,
            "firstName": "user2",
            "lastName": "tester",
            "email": "user2@dwellingly.org",
            "role": "admin",
            "archived": false,
            "properties": []
        },
        {
            "id": 3,
            "firstName": "user3",
            "lastName": "tester",
            "email": "user3@dwellingly.org",
            "role": "admin",
            "archived": false,
            "properties": []
        }
    ]
}
`

**Comment by sokampdx** [comment]
related to https://github.com/codeforpdx/dwellingly-app/issues/253

**Comment by sokampdx** [comment]
@htharker42  Should we close this after https://github.com/codeforpdx/dwellinglybackend/pull/55/files has been merged?
